Breeding-Insight/bi-api
# [BI-1818] View exp: Obs dataset tab, details, and summary stats

# Description
**Story:** [BI-1818](https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?modal=detail&selectedIssue=BI-1818)

1. Created a new controller named ListController within the org.breedinginsight.brapi.v2 package
-  Moved the getLists method from BrAPIV2Controller to this class
- Updated ListQuery to include externalReferenceId and externalReferenceSource
- Updated getLists to handle a listType of observationVariables and to pass-through externalReferenceId and externalReferenceSource, stripping the program key from the listName for each list that is returned

2. Created a Dataset model within the org.breedinginsight.model package, with fields:
- experimentId - String
- additionalInfo - JsonObject
- data - List<BrAPIObservation>
- observationUnits - List<BrAPIObservationUnits>
- observationVariables - List<BrAPIObservationVariables>

3. In ExperimentController created a method to fetch a dataset’s data for an experiment
method: GET
url: /${micronaut.bi.api.version}/programs/{programId}/experiments/{experimentId}/datasets/{datasetId}?stats=[true|false]
the result field of the response is of typeDataset

4. In BrAPITrialService created a method to fetch a dataset’s data for an experiment parameters

- programId - UUID
- experimentId - UUID
- datasetId - UUID
- includeStats - boolean
- returns → Dataset

# Dependencies
these backend changes are a dependency for [BI-1645](https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?modal=detail&selectedIssue=BI-1645)

# Testing
1. Verify that the germplam lists table still display as expected and clicking on details displays the list as expected.
2. Import an experiment with observation variables included as columns in the import. Add observation data for some of the rows of the import, but leave some rows empty, without data.
3. Using a client like curl, GET http://<local>/v1/programs/7c0d1523-0cc7-4a7f-b262-3e81746df5e7/brapi/v2/lists?externalReferenceSource=dataset&externalReferenceId=82d55336-0856-4fe2-a33d-a711a5af6bfc&listType=observationVariables, and verify the list in the response contains the observation variables used in the experiment import.
4. Using curl, GET http://<local>/v1/programs/7c0d1523-0cc7-4a7f-b262-3e81746df5e7/experiments/bdb3bd47-020a-459c-a254-5a0e062a33c2/dataset/82d55336-0856-4fe2-a33d-a711a5af6bfc?stats=true, and verify the response contains an entity of Dataset with the imported observations.
5. Compare the statistics in the additionalInfo field of the response dataset with the experiment import and confirm the counts are accurate.


# Checklist:



[BI-1818]: https://breedinginsight.atlassian.net/browse/BI-1818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BI-1645]: https://breedinginsight.atlassian.net/browse/BI-1645?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
